tensorflow_datasets/core/utils/gpath_test.py

```diff
@@ -0,0 +1,112 @@
+"""Tests for tensorflow_datasets.core.utils.gpath."""
+
+import os
+import pathlib
+from pathlib import Path
+
+import pytest
+import tensorflow as tf
+
+from tensorflow_datasets import testing
+from tensorflow_datasets.core.utils.gpath import GcsPath
+
+
```

Comment:
Also test: `str`, `os.fspath`
And make sure that `GcsPath(...).joinpath().parent` is also a `GcsPath`

Reply:
Done